test/kwok-tests/README.md

```diff
@@ -0,0 +1,207 @@
+MODIFIED   baseline-cpu-job-short-0-75crs   0/1     ContainerCreating   0          0s
+MODIFIED   baseline-cpu-job-short-0-75crs   1/1     Running             0          10s
+MODIFIED   baseline-cpu-job-short-0-75crs   0/1     Completed           0          20s
+MODIFIED   baseline-cpu-job-short-0-75crs   0/1     Completed           0          21s
+MODIFIED   baseline-cpu-job-short-0-75crs   0/1     Completed           0          22s
+```
+
+Now, we would like to simulate similar timed events for a fake job as well. KWOK's `Stage` configuration ([link](https://kwok.sigs.k8s.io/docs/user/stages-configuration/)) allows users to define and simulate different stages in the lifecycle of pods. By configuring the `delay`, `selector`, and `next` fields in a `Stage`, you can control when and how the stage is applied, providing a flexible and scalable way to simulate real-world scenarios in your Kubernetes cluster. 
+
+The YAML file [kwok.yaml](https://github.com/vishakha-ramani/multi-cluster-app-dispatcher/blob/main/test/kwok-tests/kwok.yaml) in this repo provides a custom pod lifecycle `Stage` API that simulates the events with the desired timings as described above. To run KWOK following the custom pod lifecycle, mount the configuration to `~/.kwok/kwok.yaml`, and then run KWOK controller out of your Kubernetes cluster (in my case it is a kind cluster) as follows:
+```
+kwok \
+  --kubeconfig=~/.kube/config \
+  --manage-all-nodes=false \
+  --manage-nodes-with-annotation-selector=kwok.x-k8s.io/node=fake \
+  --manage-nodes-with-label-selector= \
+  --disregard-status-with-annotation-selector=kwok.x-k8s.io/status=custom \
+  --disregard-status-with-label-selector= \
+  --cidr=10.0.0.1/24 \
+  --node-ip=10.0.0.1 \
+--config=kwok.yaml
+```
+
+*Note 1*: If you have KWOK already deployed and running inside your Kubernetes cluster, make sure to first scale it down to avoid two instances of KWOK controller running. The command
+```
+kubectl scale deployment kwok-controller --replicas=0 -n kube-system
+```
+should scale down the kwok controller in the K8 cluster. 
+
+*Note 2:* The above command essentially means that the KWOK controller will only manage nodes that have that specified annotation.
+
+Now that we have KWOK running outside the cluster with our custom `Stage` configuration, we can create a fake `Job` and deploy it on the `kind` cluster. The fake `Job` spec looks as follows:
+```
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: default
+  name: nomcadkwok-cpu-job-short-0
+spec:
+  parallelism: 1
+  completions: 1
+  template:
+    metadata:
+      namespace: default
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: type
+                    operator: In
+                    values:
+                      - kwok
+      # A taints was added to an automatically created Node.
```

Comment:
```suggestion
      # A taint was added automatically to the created Node.
```
Though the node was created **automatically** in the shell script at Line 21, the taint was also added **automatically**.